From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Substitute nimbus-fml

Adds support for using a locally built nimbus-fml binary, instead of fetching a prebuilt artifact from Mozilla.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/gradle/plugins/nimbus-gradle-plugin/src/main/kotlin/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.kt b/mobile/android/gradle/plugins/nimbus-gradle-plugin/src/main/kotlin/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.kt
index 594374187da1..d2e1fdacbcb5 100644
--- a/mobile/android/gradle/plugins/nimbus-gradle-plugin/src/main/kotlin/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.kt
+++ b/mobile/android/gradle/plugins/nimbus-gradle-plugin/src/main/kotlin/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.kt
@@ -118,7 +118,7 @@ class NimbusPlugin : Plugin<Project> {
         @Suppress("UNCHECKED_CAST")
         val mozconfigSubsts = mozconfig?.get("substs") as? Map<String, Any>
 
-        if (mozconfigSubsts?.get("MOZ_APPSERVICES_IN_TREE").isTruthy()) {
+        if (true) {
             // This is subtle.  We capture `NIMBUS_FML` in the configuration cache as a `String`.
             // If we access `project.gradle...` in the `provider` `Callable` below, we capture the
             // `Project` in the configuration cache, which is not desirable.
-- 
